hal/halx86/generic/sysinfo.c

```diff
@@ -44,7 +44,11 @@ HaliQuerySystemInformation(IN HAL_QUERY_INFORMATION_CLASS InformationClass,
 		REPORT_THIS_CASE(HalCmcLogInformation);
 		REPORT_THIS_CASE(HalCpeLogInformation);
 		REPORT_THIS_CASE(HalQueryMcaInterface);
-		REPORT_THIS_CASE(HalQueryAMLIIllegalIOPortAddresses);
+		case HalQueryAMLIIllegalIOPortAddresses:
+		{
+			/* FIXME: TODO */
+			return STATUS_NOT_IMPLEMENTED;
```

Comment:
I'm unclear on how this can fix anything. `REPORT_THIS_CASE` uses `break`, which will exit the `switch` and also return `STATUS_NOT_IMPLEMENTED`. So the only difference seems to be debug prints?

Reply:
Probably the difference is I didn't use `break` in my stub, so it fixed the problem? :thinking:

Reply:
But return code is the same anyway, so the only difference is missing `UNIMPLEMENTED` :raised_eyebrow: That's strange

Reply:
To test the hypothesis of the DPRINT vs. no DPRINT stuff @oleg-dubinskiy , I suggest you try this PR but where you add an explicit `UNIMPLEMENTED` or a `DPRINT1("Not implemented\n")` command just before your `return STATUS_NOT_IMPLEMENTED;` . If it crashes again systematically we will know that for some reason (still not understood) the DPRINT triggers the BSOD.

Reply:
No @HBelusca. :smiley: Still booting successfully for me, as with `UNIMPLEMENTED`, same with `DPRINT1("Not implemented\n")`. And what is the most surprising, is I still don't see that debug print from the log at all, in both cases.
So most of all, the problem is in something other. It seems to me that MS implementation of acpi.sys just needs a proper definition of that case, without any macros.

Reply:
But if the macro just resolves to the very exact same code as the one written in "expanded" form, this does not make any sense, and I would tend to agree with @ThFabba that it is actually a result of some sort of race condition.

Reply:
OK, now I updated the PR by the proper implementation of this, instead of stub. But it still doesn't fix CORE-17358. It even didn't improve the situation at all! :frowning_face: 
So I linked CORE-17359 instead.
Thanks to @vgalnt for the provided patch with this implementation. :+1: